python/ray/tune/experimental/output.py

```diff
@@ -766,21 +767,29 @@ def on_checkpoint(
         iteration: int,
         trials: List[Trial],
         trial: Trial,
-        checkpoint: "_TrackedCheckpoint",
+        checkpoint: Union["_TrackedCheckpoint", "Checkpoint"],
         **info,
     ):
         if self._verbosity < self._intermediate_result_verbosity:
             return
-        # don't think this is supposed to happen but just to be save.
+        # don't think this is supposed to happen but just to be safe.
         saved_iter = "?"
         if trial.last_result and TRAINING_ITERATION in trial.last_result:
             saved_iter = trial.last_result[TRAINING_ITERATION]
 
         self._start_block(f"trial_{trial}_result_{saved_iter}")
+
+        if isinstance(checkpoint, Checkpoint):
+            loc = f"({checkpoint.filesystem.type_name}){checkpoint.path}"
```

Comment:
We currently have no way to restore the actual location URI, so this is the next best thing.

Reply:
add an example of what this looks like in the pr description?